Assets/VRTK/Scripts/VRTK_RealMovement.cs

```diff
@@ -0,0 +1,350 @@
+
+        [SerializeField]
+        [Tooltip("Lower to decrease speed, raise to increase.")]
+        private float m_speedScale = 1;
+
+        [SerializeField]
+        [Tooltip("The max speed you can move in game units. (If 0 or less, max speed is uncapped)")]
+        private float m_maxSpeed = 4;
+
+        [SerializeField]
+        [Tooltip("How your movement direction will be determined.  The Gaze method tends to lead to the least motion sickness.  Smart decoupling is still a Work In Progress.")]
+        private DirectionalMethod m_dirMethod = DirectionalMethod.Gaze;
+
+        [SerializeField]
+        [Tooltip("If we're using Smart Decoupling, all trackables must be within this degree threshold to change direction.")]
+        private float m_smartDecoupleThreshold = 30f;
+
+        // The cap before we stop adding the delta to the movement list.  This will help regulate speed.
+        [SerializeField]
+        [Tooltip("The max amount of movement we wish to register in the virutal world.  Decreasing this will increase acceleration, and vice versa.")]
+        private float m_sensitivity = 0.02f;
+
+        private int m_averagePeriod = 60;
+        private float m_cameraLastYRotation = 0f;
+        
+        // Which tracked objects to use to determine amount of movement.
+        private List<Transform> m_trackedObjects = new List<Transform>();
+
+        private VRTK_PlayerPresence m_playerBody;
+        private Rigidbody m_rigidBody;
+        private GameObject controllerLeftHand;
+        private GameObject controllerRightHand;
+        private Transform headset;
+        private bool leftSubscribed;
+        private bool rightSubscribed;
+        private ControllerInteractionEventHandler touchpadPressed;
+        private ControllerInteractionEventHandler touchpadUp;
+
+        // List of all the frames movements over the average period.
+        private Dictionary<Transform, List<float>> m_movementList = new Dictionary<Transform, List<float>>();
+        private Dictionary<Transform, float> m_previousYPositions = new Dictionary<Transform, float>();
+
+        private float m_playAreaPreviousY = 0f;
+
+        private Vector3 m_initalGaze = Vector3.zero;
+
+        private Dictionary<Transform, Vector3> m_initialTrackableDirs = new Dictionary<Transform, Vector3>();
+
+        private float m_curSpeed = 0.00f;
+
+        private Vector3 m_direction = new Vector3();
+
+        private bool active = false;
+
+        // If we're in Unity 5.4, we need to make sure we are using the eyes, not the head.
+        void Awake()
+        {
+            controllerLeftHand = VRTK_DeviceFinder.GetControllerLeftHand();
+            controllerRightHand = VRTK_DeviceFinder.GetControllerRightHand();
+            headset = VRTK_DeviceFinder.HeadsetTransform();
+
+            setControlOptions(m_controlOptions);
+
+            touchpadPressed = new ControllerInteractionEventHandler(DoTouchpadDown);
```

Comment:
you should probably do things like this in the `OnEnable` then you can deregister events in the `OnDisable`